src/App.tsx

```diff
@@ -15,49 +16,128 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((name1, name2) => (
+        name1.localeCompare(name2)
+      ));
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((product1, product2) => (
+        product1.length - product2.length
+      ));
+      break;
+
+    default: break;
```

Comment:
what about SortType.NONE ?
```suggestion
    case SortType.NONE:
    default: break;
```